dataland-frontend/src/components/general/SfdrDataTable.vue

```diff
@@ -71,24 +71,30 @@ import DataTable from "primevue/datatable";
 import Column from "primevue/column";
 import DetailsCompanyDataTable from "@/components/general/DetailsCompanyDataTable.vue";
 import { lksgModalColumnHeaders } from "@/components/resources/frameworkDataSearch/lksg/LksgModalColumnHeaders";
-import { ReportingPeriodOfDataSetWithId } from "@/utils/DataTableDisplay";
+import {
+  ReportingPeriodOfDataSetWithId,
+  mountRowHeaderClickEventListeners,
+  unmountRowHeaderClickEventListeners,
+} from "@/utils/DataTableDisplay";
+import { SfdrKpiObject } from "@/components/resources/frameworkDataSearch/sfdr/SfdrPanel.vue";
 
 export default defineComponent({
-  name: "CompanyDataTable",
+  name: "SfdrDataTable",
   components: { DataTable, Column },
   directives: {
     tooltip: Tooltip,
   },
   data() {
     return {
-      kpiDataObjectsToDisplay: [],
+      kpiDataObjectsToDisplay: [] as SfdrKpiObject[],
       expandedRowGroups: ["_general"],
       listOfProductionSitesConvertedNames: lksgModalColumnHeaders.listOfProductionSites,
+      rowClickHandlersMap: new Map() as Map<Element, EventListener>,
```

Comment:
Isn't this always empty?  I can't find code that actually puts values into this map.

Reply:
Actually, that is a very good point. I need to check if we're correctly removing event listeners and I suspect that at the moment we are not... and that's why the `return handlerMap` was important in `DataTableDisplay.ts`. It returned the list of event listeners that was later used to unmount and destroy event listeners. I'll check.

Reply:
Ok then you will need to store the handlerMap in the respective vue components, to then remove the listeners on unmount. This storage is currently not happening.

Reply:
Exactly. Working on fix right now. Will need to move the setTimeout() to component where this is being mounted. Otherwise it just returns an empty array.

settimout() is needed to wait an extra tick after the page is rendered. No delay is set so it is the default 0ms. Without this no elements are parsed ;)

Reply:
Done.

Reply:
Have you tried putting `mountRowHeaderClickEventListeners` into the "mounted()" block of the component?
Then the page is already rendered, and it might be possible to remove the "setTimeout"-wrapper.

Reply:
Yes, but it didn't work either. only within setTimeout().. either created or mounted

Reply:
Ok.